assets/cluster-monitoring-operator/prometheus-rule.yaml

```
@@ -409,6 +409,21 @@ spec:
       labels:
         workload_type: deploymentconfig
       record: namespace_workload_pod:kube_pod_owner:relabel
+  - name: openshift-etcd-telemetry.rules
+    rules:
+    - expr: sum by (instance) (etcd_mvcc_db_total_size_in_bytes{job="etcd"})
+      record: cluster:etcd_mvcc_db_total_size_in_bytes:sum
+    - expr: histogram_quantile(0.99, sum(rate(etcd_disk_wal_fsync_duration_seconds_bucket{job="etcd"}[5m]))
+        by (instance, le))
+      record: histogram_quantile_99:etcd_disk_wal_fsync_duration_seconds_bucket:sum
```

For recording rules based on histogram quantiles, we usually add generic names to the recording rules along with quantile label that will distinguish the different percentile that can be computed via this recording rule. This makes it easier to extend the recording rule if we need to compute more percentiles at some point.
```suggestion
      labels:
        quantile: "0.99"
      record: etcd_quantile:etcd_disk_wal_fsync_duration_seconds:histogram_quantile
```

---

Yes I know, but wasn't sure if that is possible to be sent via telemetry or in which format?

---

I'm not aware of any limitations on the telemetry side that would prevent metrics from being sent.

---

So we just allowlist this rule and any label matches would be sent.

---

Hmm good point, I think that if we were to add the label to the telemeter query today, it would apply to all the metrics as `alertstate="firing"` currently. As far as I understand it, labels are currently global to the telemeter query, so if we were to add recording rules with different quantiles later, we wouldn't be able to select only one to send via telemetry.

---

Yes, not sure how much of a win defining them this way would be as I remember we had some issues on the label side in the past. But up to you two really, happy to go any route there.

---

> So we just allowlist this rule and any label matches would be sent. 

We could narrow-down the selected series in `manifests/0000_50_cluster-monitoring-operator_04-config.yaml` to sent only the 99th percentile value (e.g. `{__name__="etcd_quantile:etcd_disk_wal_fsync_duration_seconds:histogram_quantile",quantile="0.99"}`) but I'm not sure it's worth the effort.

---

Yes exactly my thinking. I also feel like querying might be more awkward as majority rules are sent directly.

---

I would personally prefer having a generic recording rule narrowed down in telemetry. If we ever need to extend the recording rule to get another percentile, we would be able to keep it under the same recording rule which would be easier to manipulate and query.

---

Can we add that when we extend the recording rules, otherwise I fear premature optimization here? (We are already at our budget right now, so we most likely won't add it, is why I am asking :) )

---

It doesn't have to be a recording rule sent via telemetry necessarily.

I essentially just want to make sure that we stick to the original format that we used for the other quantile recording rules. If y'all think we should make an exception here, I am fine with it.